lib/simplecov/exit_codes/maximum_coverage_drop_check.rb

```diff
@@ -36,14 +38,24 @@ def last_run
         @last_run = SimpleCov::LastRun.read
       end
 
-      def coverage_diff
-        raise "Trying to access coverage_diff although there is no last run" unless last_run
-
-        @coverage_diff ||= last_run[:result][:covered_percent] - covered_percent
+      def coverage_drop_violations
+        @coverage_drop_violations ||=
+          compute_coverage_drop_violations.select do |achieved|
+            achieved.fetch(:max_drop) < achieved.fetch(:drop_percent)
+          end
       end
 
-      def covered_percent
-        SimpleCov.round_coverage(result.covered_percent)
+      def compute_coverage_drop_violations
+        maximum_coverage_drop.map do |criterion, percent|
+          {
+            criterion: criterion,
+            max_drop: percent,
+            drop_percent: last_run[:result][criterion] -
```

Comment:
This is probably the place that'd blow up/we need to fix to be compatible with old `last_run.json` files. This would be `nil` then and we'd get no `-` for `nil:NilClass`